### PR TITLE
WT-8624 Output perf plots in Evergreen from cpp suite tests

### PR DESCRIPTION
**Pull request by etienneptl**
The main changes are:

- New configuration item `save` that can be added to a statistics in a configuration file to indicate whether we want to save the associated statistics values in a JSON file.
- A JSON file is generated at the end of each cpp suite tests with the statistic values that were saved.
- For each test from the cpp suite, the aforementioned JSON file is pushed to Evergreen so we can visualize the trend of each stat for each test.

### CODE REVIEW

File: test/cppsuite/configs/hs_cleanup_default.txt [hunk 1]
```diff
@@ -19,22 +19,25 @@ runtime_monitor=
...
     cc_pages_removed=
     (
         max=10000000,
         postrun=true,
+        save=true,
     ),
     stat_cache_size=
     (
-        max=1000,
+        max=100,
```
Comment (ajmorton) [inline test/cppsuite/configs/hs_cleanup_default.txt:32]: Is this change intentional? It seems unrelated to stat reporting

File: test/evergreen.yml [hunk 2]
```diff
@@ -1048,6 +1068,9 @@ tasks:
         vars:
           test_config_filename: configs/base_test_stress.txt
           test_name: base_test
+      - func: "upload cpp test stats"
+        vars:
+          test_name: base_test
```
Comment (ajmorton) [inline test/evergreen.yml:1073]: Both `cppsuite-base-test-stress` and `cppsuite-hs-cleanup-default` are uploading a file called `base_test.json`. To be safe I checked that one test output doesn't overwrite the other and things are ok 👍 .

File: test/cppsuite/test_harness/runtime_monitor.cxx [hunk 3]
```diff
@@ -338,4 +351,30 @@ runtime_monitor::finish()
...
+ */
+void
+runtime_monitor::save_stats(const std::string &filename)
+{
+    std::string stat_info;
+    std::ofstream file(filename + ".json");
+    file << "[{\"info\":{\"test_name\": \"" << filename << "\"},\"metrics\": [";
+
+    for (const auto &stat : _stats) {
+        if (stat->get_save())
```
Comment (ajmorton) [inline test/cppsuite/test_harness/runtime_monitor.cxx:366]: `get_save` is a bit clunky here. 
It doesn't match the traditional getter naming pattern, but considering something like `should_save` instead.

**Review by ajmorton — APPROVED** [review]
LGTM!
A few clarifying questions but nothing blocking.

File: test/evergreen.yml [hunk 4]
```diff
@@ -575,6 +575,11 @@ functions:
         set -o verbose
         ${test_env_vars|} test/csuite/${test_binary}/smoke.sh ${test_args|} 2>&1
 
+  "upload cpp test stats":
+    - command: perf.send
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply test/evergreen.yml:579]: Yes, see WT-8685

File: test/cppsuite/configs/hs_cleanup_default.txt (hunk 1 above)
Comment (etienneptl) [reply test/cppsuite/configs/hs_cleanup_default.txt:32]: It is intentional but I am happy to revert it. I am pretty sure it is a mistake that I spotted when I was reviewing the stats.

File: test/evergreen.yml (hunk 2 above)
Comment (etienneptl) [reply test/evergreen.yml:1073]: Thanks for checking @ajmorton !

File: test/cppsuite/test_harness/runtime_monitor.cxx (hunk 3 above)
Comment (etienneptl) [reply test/cppsuite/test_harness/runtime_monitor.cxx:366]: As discussed, I would like to keep the `get_xx` pattern for getters.

File: test/cppsuite/configs/hs_cleanup_default.txt (hunk 1 above)
Comment (ajmorton) [reply test/cppsuite/configs/hs_cleanup_default.txt:32]: No worries then, just wanted to confirm it wasn't accidental

File: test/cppsuite/test_harness/runtime_monitor.cxx [hunk 5]
```diff
@@ -338,4 +351,30 @@ runtime_monitor::finish()
...
+            stat_info += "{\"name\":\"" + stat->get_name() +
```
Comment (AlexeyAnisimovAU) [inline test/cppsuite/test_harness/runtime_monitor.cxx:367]: Monir note: why do we need `stat_info` if we already opened `file` stream and started pushing data in it? We can either use the string buffer or the file stream for all the data.

Comment (etienneptl) [reply test/cppsuite/test_harness/runtime_monitor.cxx:367]: We could use the file stream directly indeed. But the annoying part is the extra comma at the end that is removed lin 373. That's why I wanted a temporary `string` so I can edit it before writing it into the file.

Comment (AlexeyAnisimovAU) [reply test/cppsuite/test_harness/runtime_monitor.cxx:367]: You could use the string buffer to build JSON, create the file stream and flush the data at the very end. Or use the stream for everything add logic to add a comma at the beginning of the `for` loop.

Comment (etienneptl) [reply test/cppsuite/test_harness/runtime_monitor.cxx:365]: I used a `string` for the entire buffer before adding it to the file, see 5be76e2d74b355999f0dc6bbd98790ea3ba8bb5b. Thanks for this good suggestion!

File: test/cppsuite/test_harness/runtime_monitor.cxx [hunk 6]
```diff
@@ -338,4 +351,29 @@ runtime_monitor::finish()
...
+    std::string stat_info = "[{\"info\":{\"test_name\": \"" + filename + "\"},\"metrics\": [";
+
+    for (const auto &stat : _stats) {
+        if (stat->get_save())
+            stat_info += "{\"name\":\"" + stat->get_name() +
+              "\",\"value\":" + stat->get_value_str(_cursor) + "},";
+    }
+
+    /* Remove last extra comma. */
+    if (!stat_info.empty()) {
```
Comment (etienneptl) [inline test/cppsuite/test_harness/runtime_monitor.cxx:370]: I will need to fix an error here as it is always true, thanks @ajmorton for spotting it.

Comment (etienneptl) [reply test/cppsuite/test_harness/runtime_monitor.cxx]: Fixed in 2251c4838de185c84ab00cf4485daf64c137da8f

**Review by Siddhartha8899 — APPROVED** [review]
LGTM

**Review by AlexeyAnisimovAU — APPROVED** [review]
LGTM!
